tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+YML structure:
+```yml
+add-paths:                 # Start a list path names that should be added to the include file search
+  - path                   # add path name
+  - path
+```
+
+## Del-Paths
+
+Remove include paths (that are defined at the cproject level) from the command line of the development tools.
+
+YML structure:
+```yml
+del-paths:                 # Start a list of path names that should be removed from the include file search
+  - path                   # remove path name
+  - *                      # remove all paths
+```
+
+
+
+## Misc
+
+Add tool specific controls as literal strings that are directly passed to the individual tools.
+
+YML structure:
+```yml
+misc:                      # Start a list of literal control strings that are directly passed to the tools.
+  - compiler:              # select the toolchain that the literal control string applies too (AC6, IAR, GCC).
+    C: string              # applies to *.c files only.
+    CPP: string            # applies to *.cpp files only.
+    C*: string             # applies to *.c and *.cpp files.
+    ASM: string            # applies to assembler source files
+    Link: string           # applies to the linker
+    Lib: string            # applies to the library manager or archiver
+```
+
+
+## Solution: Collection of related Projects
+
+The section [Project setup for related projects](#project-setup-for-related-projects) describes the collection of related projects.  The file `*.csolution.yml` describes the relationship of this projects.  This file may also define [Target and Build Types](#target-and-build-types) before the section `solution:`.
+
+The YML structure of the section `solution:` is:
+
+```yml
+solution:                  # Start a list of projects.
+  - project:               # path to the project file (required).
+    for-type:              # include project for a list of *build* and *target* types (optional).
+    not-for-type:          # exclude project for a list of *build* and *target* types (optional).
+    compiler:              # specify a specific compiler
+    optimize:              # optimize level for code generation (optional)
+    debug:                 # generation of debug information (optional)
+    defines:               # define symbol settings for code generation (optional).
+    undefines:             # remove define symbol settings for code generation (optional).
+    add-paths:             # additional include file paths (optional).
+    del-paths:             # remove specific include file paths (optional). 
+    misc:                  # Literal tool-specific controls
+```
+
+## Project Definition
+
+Start of a project definition in a `*.cproject.yml` file (optional section)
+
+The YML structure of the section `project:` is:
+```yml
+project:                     # do we need this section at all, perhaps for platform?
+  compiler: name             # specify compiler (AC6, GCC, IAR)
```

Comment:
So each project would support only 1 compiler ?

Reply:
No, this example just shows the compiler selection at 'project:' level.  However, you can select the 'compiler' also at solution, build-type, target-type level. Take a look to "General Properties".